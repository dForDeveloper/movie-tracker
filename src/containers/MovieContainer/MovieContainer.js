import React from 'react';
import { connect } from 'react-redux';

const MovieContainer = ({ movies }) => {
  return (
    <div>
      {
        movies.map(movie => {
          return <p key={movie.id}>{movie.title}</p>
        })
      }
    </div>
  )

}

const mapStateToProps = (state) => {
  return { movies: state.movies }
}

export default connect(mapStateToProps)(MovieContainer);