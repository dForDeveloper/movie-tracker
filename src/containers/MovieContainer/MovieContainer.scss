.MovieContainer {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  align-items: center;
  justify-items: center;
  border-right: 3px solid $border-color;
  border-left: 3px solid $border-color;
}