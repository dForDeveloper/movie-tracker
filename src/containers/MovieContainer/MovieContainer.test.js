import React from 'react';
import { shallow } from 'enzyme';
import MovieContainer from './MovieContainer';

describe('MovieContainer', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <MovieContainer />
    );
  });

  describe('MovieContainer container', () => {
    it('should properly render the component elements', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it.skip('should return all matching movies when getMoviesToDisplay is invoked', () => {
      const mockMovies = [{id: 1, favorite: false}];
      const expected = wrapper.instance().getMoviesToDisplay(mockMovies);
      expect(mockMovies).toMatchObject(expected);
    });
  
    it.skip('should return all matching favorite movies by their id when getMoviesToDisplay is invoked', () => {
      const mockMovies = [{id: 1, favorite: false}, {id: 2, favorite: true}];
      const expected = wrapper.instance().getMoviesToDisplay(mockMovies);
      expect(mockMovies).toMatchObject(expected);
    });
  });

  describe('mapStateToProps', () => {});
  
});