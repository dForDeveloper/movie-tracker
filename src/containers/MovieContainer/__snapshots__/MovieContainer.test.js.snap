// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer MovieContainer container should match snapshot when path is / and showLoginPrompt is false 1`] = `
<div
  className="MovieContainer"
>
  <Connect(Header) />
  <Connect(MovieCard)
    favorite={false}
    id={1}
    key="1"
  />
  <Connect(MovieCard)
    favorite={false}
    id={2}
    key="2"
  />
</div>
`;
